### ISSUE TITLE
Realm List<String> fails to map in toDictionary(), so when passed to realm create method, crashes

### ISSUE THREAD

**Issue by akramhussein** [evermeer/EVReflection#277]
Using the following model, when I convert an object to an NSDictionary and insert using realm's `create` method, I receive a crash as seen below.

I've tried:

* Adding a `skipProperty` step to check count of 0 on NSArray, NSDictionary, [String].
* Tried to map the field with a `propertyConverters` but it never gets called

**Model**

```
import Foundation
import RealmSwift
import EVReflection

class Base: Object, EVReflectable {
    
    @objc dynamic var id: String = ""
   
    override static func primaryKey() -> String? { return "id" }
}

final class Person: Base {
    
    @objc dynamic var name: String = ""

    let aliases = List<String>()
    
    override static func primaryKey() -> String? { return "id" }

}
```
**Error**:

```
reason: 'Invalid value '{
}' of type '__NSDictionary0' for 'string' array property 'Person.aliases'.'
*** First throw call stack:
```

Seems its trying to parse an empty array, which is being stored as an empty NSDictionary.

**Comment by akramhussein** [comment]
I've checked and `constructWith` in `extension List : EVCustomReflectable` from file [RealmListEVCustomReflectable.swift)](https://github.com/evermeer/EVReflection/blob/fa65a1a26f2065becb6c30cc6ff300855986912f/Source/Realm/RealmListEVCustomReflectable.swift) is called.

**Comment by evermeer** [comment]
The error comes from the function RLMArrayValidateMatchingObjectType in RLMArray.
As you can see in the error message the value is empty but it is a NSDictionary.
I was able to reproduce the issue.
If you have a List<Strings> and to a .toDictionary() then EVReflection does not recognize it as a list of primitive types. Instead it tries to create a dictionary for every item. If you then try to create an object from that same dictionary, then you will get this error.

I will have a look if I can fix the .toDictionary() for List<String> fields.

**Comment by evermeer** [comment]
Ah, there was a bug in the toCodableValue of RealmListEVReflectable.swift
I will publish the fix now. It should be available in about an hour.

**Comment by evermeer** [comment]
It's published as version 5.5.6

**Comment by akramhussein** [comment]
Thank you so much - this fixed the issue 👍
